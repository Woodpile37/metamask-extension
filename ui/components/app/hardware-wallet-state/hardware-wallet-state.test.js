import React from 'react';
import { act } from '@testing-library/react-hooks';
import configureMockStore from 'redux-mock-store';
import { cloneDeep } from 'lodash';

import { renderWithProvider } from '../../../../test/lib/render-helpers';
import mockState from '../../../../test/data/mock-state.json';
import { HardwareWalletStates } from '../../../../shared/constants/hardware-wallets';
import HardwareWalletState from '.';

const customStore = ({ hardwareWalletState } = {}) => {
  const data = cloneDeep({
    ...mockState,
    appState: {
      ...mockState?.appState,
      hardwareWalletState,
    },
  });
  return configureMockStore()(data);
};

describe('HardwareWalletState Component', () => {
  it('should match default snapshot', () => {
    const { container } = renderWithProvider(
      <HardwareWalletState />,
      customStore(),
    );

    expect(container).toMatchSnapshot();
  });

  it('should render message with locked state', () => {
    const { container } = renderWithProvider(
      <HardwareWalletState />,
      customStore({ hardwareWalletState: HardwareWalletStates.locked }),
    );

    expect(container).toMatchSnapshot();
  });

  it('should render empty with unlocked state', () => {
    const { container } = renderWithProvider(
      <HardwareWalletState />,
      customStore({ hardwareWalletState: HardwareWalletStates.unlocked }),
    );

    expect(container).toMatchSnapshot();
  });

  it('should render empty when headless with locked state', () => {
    const { container } = renderWithProvider(
      <HardwareWalletState headless />,
      customStore({ hardwareWalletState: HardwareWalletStates.locked }),
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onUpdate handler', async () => {
    let status = HardwareWalletStates.unlocked;
    const handler = (state) => {
      status = state;
    };
    const props = {
      pollingRateMs: 600,
      onUpdate: handler,
    };
    const { container } = renderWithProvider(
      <HardwareWalletState {...props} />,
      // start unlocked and wait to update to locked
      customStore({ hardwareWalletState: status }),
    );
    // should not change until polling commences
    expect(status).toStrictEqual(HardwareWalletStates.unlocked);

    // wait for polling to complete (allow time for at least 2 polling cycles)
    await act(() => new Promise((res) => setTimeout(res, 1500)));
    expect(status).toStrictEqual(HardwareWalletStates.locked);
    expect(container).toMatchSnapshot();
  });
});
