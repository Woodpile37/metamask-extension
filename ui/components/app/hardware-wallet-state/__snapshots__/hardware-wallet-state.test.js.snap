// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HardwareWalletState Component should call onUpdate handler 1`] = `
<div>
  <div
    class="box mm-banner-base mm-banner-alert mm-banner-alert--severity-warning box--padding-3 box--padding-left-2 box--display-flex box--gap-2 box--flex-direction-row box--background-color-warning-muted box--rounded-sm"
  >
    <span
      class="box mm-icon mm-icon--size-lg box--display-inline-block box--flex-direction-row box--color-warning-default"
      style="mask-image: url('./images/icons/warning.svg');"
    />
    <div>
      <p
        class="box mm-text mm-text--body-md box--flex-direction-row box--color-text-default"
      >
        Cannot connect to Ledger device. Please make sure your device is unlocked and Ethereum app is opened.
      </p>
    </div>
  </div>
</div>
`;

exports[`HardwareWalletState Component should match default snapshot 1`] = `<div />`;

exports[`HardwareWalletState Component should render empty when headless with locked state 1`] = `<div />`;

exports[`HardwareWalletState Component should render empty with unlocked state 1`] = `<div />`;

exports[`HardwareWalletState Component should render message with locked state 1`] = `
<div>
  <div
    class="box mm-banner-base mm-banner-alert mm-banner-alert--severity-warning box--padding-3 box--padding-left-2 box--display-flex box--gap-2 box--flex-direction-row box--background-color-warning-muted box--rounded-sm"
  >
    <span
      class="box mm-icon mm-icon--size-lg box--display-inline-block box--flex-direction-row box--color-warning-default"
      style="mask-image: url('./images/icons/warning.svg');"
    />
    <div>
      <p
        class="box mm-text mm-text--body-md box--flex-direction-row box--color-text-default"
      >
        Cannot connect to Ledger device. Please make sure your device is unlocked and Ethereum app is opened.
      </p>
    </div>
  </div>
</div>
`;
